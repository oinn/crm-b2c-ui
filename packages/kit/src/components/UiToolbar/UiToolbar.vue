<template>
  <div :class="classes">
    <slot></slot>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue';
import { themeable } from '@/mixins';

export default defineComponent({
  name: 'UiToolbar',
  mixins: [themeable],
  props: {
    app: {
      type: Boolean as PropType<boolean>,
      default: false,
    },
    dense: {
      type: Boolean as PropType<boolean>,
      default: false,
    },
    rarefied: {
      type: Boolean as PropType<boolean>,
      default: false,
    },
  },
  computed: {
    classes() {
      return {
        'ui-toolbar': true,
        ...this.themeClasses,
        'ui-toolbar--app': this.app,
        'ui-toolbar--dense': this.dense,
        'ui-toolbar--rarefied': this.rarefied,
      };
    },
  },
  mounted() {
    this.$ui.uiMainOffset.top = this.app ? this.$el.clientHeight : 0;
  },
});
</script>

<style lang="scss">
// noinspection CssUnknownTarget
@import "@/styles/config.scss";
// noinspection CssUnknownTarget
@import "@/styles/colors/index.scss";

.ui-toolbar {
  padding: 0 4 * $module;
  height: 48px;
  display: flex;
  flex-direction: row;
  align-items: center;

  color: inherit;

  &.ui-theme--light {
    background-color: get-element-color("light", "background");

    color: get-element-color("light", "text-primary")
  }
  &.ui-theme--dark {
    background-color: get-element-color("dark", "background");

    color: get-element-color("dark", "text-primary")
  }

  &.ui-toolbar--app {
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
  }

  &.ui-toolbar--dense {
    height: 36px;
  }

  &.ui-toolbar--rarefied {
    height: 56px;
  }
}
</style>
