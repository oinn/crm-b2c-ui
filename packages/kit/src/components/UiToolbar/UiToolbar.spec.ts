import { mount, MountingOptions, VueWrapper } from '@vue/test-utils';
import UiToolbar from './UiToolbar.vue';

type Props = InstanceType<typeof UiToolbar>['$props'];

describe('UiToolbar', () => {
  // Preparation
  let mountFunction: (options?: Record<string, unknown>) => VueWrapper;
  beforeEach(() => {
    mountFunction = (options = {}) => mount(UiToolbar, { ...options });
  });

  it('Rendered correctly', () => {
    const wrapper = mountFunction({
      slots: {
        default: 'Default slot text',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Contains correct class when it has app property', () => {
    const wrapper = mountFunction({
      propsData: {
        app: true,
      } as MountingOptions<Props>,
    });

    expect(wrapper.classes()).toContain('ui-toolbar--app');
  });

  it('Contains correct class when it has dense property', () => {
    const wrapper = mountFunction({
      propsData: {
        dense: true,
      } as MountingOptions<Props>,
    });

    expect(wrapper.classes()).toContain('ui-toolbar--dense');
  });

  it('Contains correct class when it has rarefied property', () => {
    const wrapper = mountFunction({
      propsData: {
        rarefied: true,
      } as MountingOptions<Props>,
    });

    expect(wrapper.classes()).toContain('ui-toolbar--rarefied');
  });

  // Don't have test for check rewriting this.$ui.uiMainOffsets global parameter.
});
