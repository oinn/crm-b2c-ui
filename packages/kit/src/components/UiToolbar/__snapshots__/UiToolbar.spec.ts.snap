// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiApp Rendered correctly 1`] = `"<div class=\\"ui-toolbar ui-theme--light\\">Default slot text</div>"`;
